File: snippets/php/simple-post.php
$payload = http_build_query([
  "key1" => "value1",
  "key2" => "value2"
]);

$options = [
  "http" => [
    "method" => "POST",
    "header" => "Content-type: application/x-www-form-urlencoded",
    "content" => $payload,
    "ignore_errors" => true,
  ]
];

$response = file_get_contents(
  "http://www.example.com",
  false,
  stream_context_create($options)
);

File: snippets/java/simple-post.java
import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpRequest.BodyPublisher;
import java.net.http.HttpRequest.BodyPublishers;
import java.net.http.HttpResponse;
import java.net.http.HttpResponse.BodyHandlers;

public class Program {

  public static void main(String[] args) throws Exception {

    HttpClient client;
    HttpRequest request;
    HttpResponse<String> response;

    BodyPublisher payload = BodyPublishers
                              .ofString("key1=value1&key2=value2");

    client = HttpClient
               .newBuilder()
               .version(HttpClient.Version.HTTP_1_1)
               .build();

    request = HttpRequest
                .newBuilder()
                .POST(payload)
                .uri(URI.create("http://example.com"))
                .header("Content-Type", "application/x-www-form-urlencoded")
                .build();

    response = client.send(request, BodyHandlers.ofString());

    String content = response.body();

  }

}
